force_torque_sensor_controller/src/force_torque_sensor_controller.cpp

```diff
@@ -0,0 +1,171 @@
+  if (sensor_name_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'sensor_name' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  frame_id_ = node_->get_parameter("frame_id").as_string();
+  if (frame_id_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'frame_id' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  fx_range_ = node_->get_parameter("fx_range").as_int();
+  if (fx_range_ == -1) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'fx_range' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  tz_range_ = node_->get_parameter("tz_range").as_int();
+  if (tz_range_ == -1) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'tz_range' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  RCLCPP_INFO_STREAM(get_node()->get_logger(), "configure successful");
+
+  try {
+    // register ft sensor data publisher
+    sensor_state_publisher_ = get_node()->create_publisher<geometry_msgs::msg::WrenchStamped>(
+        "force_torque_sensor", rclcpp::SystemDefaultsQoS());
+  } catch (...) {
+    return CallbackReturn::ERROR;
+  }
+
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorController::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration command_interfaces_config;
+  command_interfaces_config.type = controller_interface::interface_configuration_type::NONE;
+
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorController::state_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+  state_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  for (const auto & interface : state_interface_names_) {
+    state_interfaces_config.names.push_back(sensor_name_ + "/" + interface);
+  }
+
+  return state_interfaces_config;
+}
+
+CallbackReturn ForceTorqueSensorController::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn ForceTorqueSensorController::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type ForceTorqueSensorController::update()
+{
+  geometry_msgs::msg::Vector3 f_vec;
+  geometry_msgs::msg::Vector3 t_vec;
+
+  if (state_interfaces_.size() != state_interface_names_.size())
+    return controller_interface::return_type::ERROR;
+
+  for (auto index = 0ul; index < state_interfaces_.size(); ++index) {
+    switch (index) {
+      case 0:
+        f_vec.set__x(state_interfaces_[index].get_value());
+        break;
+      case 1:
+        t_vec.set__z(state_interfaces_[index].get_value());
+        break;
+      default:
+        break;
+    }
+  }
+
+  wrench_state_msg_.header.stamp = get_node()->get_clock()->now();
+  wrench_state_msg_.header.frame_id = frame_id_;
```

Comment:
add this into configure.

Reply:
I think, line number 153 can be added inside configure. But line 152 should be here as it will be different every time it publishes the message. Please suggest.